packages/create-sitecore-jss/src/templates/nextjs-sxa/src/Layout.tsx

```diff
@@ -37,19 +37,29 @@ const Layout = ({ layoutData, headLinks }: LayoutProps): JSX.Element => {
           <link rel={headLink.rel} key={headLink.href} href={headLink.href} />
         ))}
       </Head>
-
       {/* root placeholder for the app, which we add components to using route data */}
-      <div className={mainClassPageEditing}>
-        <header>
-          <div id="header">{route && <Placeholder name="headless-header" rendering={route} />}</div>
-        </header>
-        <main>
-          <div id="content">{route && <Placeholder name="headless-main" rendering={route} />}</div>
-        </main>
-        <footer>
-          <div id="footer">{route && <Placeholder name="headless-footer" rendering={route} />}</div>
-        </footer>
-      </div>
+      <% if (xmcloud) {-%>
+      {layoutData.sitecore.context.renderingType === RenderingType.Component ? (
```

Comment:
After having a second thought, I don't really like mixing of component library functionality with Layout. It creates unclear responsibilities, and makes updgrading procedure harder and give no control on final output in case of library.
For now I have two alternatives in mind with changes moved to catch all route:
1. Projecting layout into component and passing an updated layout with specified target placeholder.
```jsx
<ComponentLibrary layoutData={...layoutData} targetPlaceholder="headless-main">
(componentLayout)=> {<Layout layoutData/>}
</ComponentLibrary>
```
2. Having a default layout specified in comp `ComponentLibrary` with all required things like headlinks and scripts + provide optional props to provide a custom one